src/content/en/updates/2017/02/media-session.md

```diff
@@ -0,0 +1,374 @@
+      event.waitUntil(initArtworksCache());
+    });
+    
+    function initArtworksCache() {
+      caches.open('artworks-cache-v1')
+      .then(cache => cache.add(FALLBACK_ARTWORK_URL));
+    }
+    
+    self.addEventListener('fetch', event => {
+      if (/artwork-[0-9]+\.png$/.test(event.request.url)) {
+        event.respondWith(handleFetchArtwork(event.request));
+      }
+    });
+    
+    function handleFetchArtwork(request) {
+      // Return cache request if it's in the cache already, otherwise fetch
+      // network artwork.
+      return getCacheArtwork(request)
+      .then(cacheResponse => cacheResponse || getNetworkArtwork(request));
+    }
+    
+    function getCacheArtwork(request) {
+      return caches.open('artworks-cache-v1')
+      .then(cache => cache.match(request));
+    }
+    
+    function getNetworkArtwork(request) {
+       // Fetch network artwork.
+       return fetch(request)
+       .then(networkResponse => {
+         if (networkResponse.status !== 200) {
+           return Promise.reject('Network artwork response is not valid');
+         }
+         // Add artwork to the cache for later use and return network response.
+         addArtworkToCache(request, networkResponse.clone())
+         return networkResponse;
+       })
+       .catch(error => {
+         // Return cached fallback artwork.
+         return getCacheArtwork(new Request(FALLBACK_ARTWORK_URL))
+       });
+    }
+    
+    function addArtworkToCache(request, response) {
+      return caches.open('artworks-cache-v1')
+      .then(cache => cache.put(request, response));
+    }
+
+Caution: If you want your service worker to be able to intercept artwork
+network requests on [the very first page load], you may want to call
+`clients.claim()` within your service worker once it's activated.
+
+### Let user control cache
+
+As user will consume content from your web app, media files and artworks may
+take a lot of space on user's device. It is **your responsibility to show how
+much cache is used and give user the ability to clear it**. Thankfully for us,
+doing so is pretty easy with the [Cache API].
+
+    // Here's how I'd compute how much cache is used by artworks... 
+    caches.open('artworks-cache-v1')
+    .then(cache => cache.matchAll())
+    .then(responses => Promise.all(responses.map(r => r.blob())))
+    .then(blobs => blobs.map(blob => blob.size).reduce((acc, cur) => acc + cur, 0))
+    .then(cacheSize => { console.log('Artworks cache is ' + cacheSize + ' bytes.'); })
+    .catch(error => { console.log(error); });
+
+    // And here's how to delete some artworks...
+    const artworksToDelete = ['artwork1.png', 'artwork2.png', 'artwork3.png'];
+
+    caches.open('artworks-cache-v1')
+    .then(cache => Promise.all(artworksToDelete.map(artwork => cache.delete(artwork))))
+    .catch(error => { console.log(error); });
+
+## Implementation nits
+
+- Chrome Media notifications show only if media files duration is [at least 5 seconds].
```

Comment:
In case of, we don't take full audio focus if shorter than 5 seconds. We take the intermittent one.